llvm/lib/Target/AMDGPU/GCNSubtarget.cpp

```diff
@@ -771,6 +775,9 @@ GCNUserSGPRUsageInfo::GCNUserSGPRUsageInfo(const Function &F,
 
   if (hasPrivateSegmentSize())
     NumUsedUserSGPRs += getNumUserSGPRForField(PrivateSegmentSizeID);
+
+  if (hasLDSKernelId())
+    NumSyntheticSGPRs += getNumUserSGPRForField(LDSKernelIdID);
```

Comment:
I'm not sure I understand why anything is changing with the kernel ID. Is this just changing bookkeeping to split out the kernel's user SGPRs. vs. the implicit SGPRs in a function?